erizoAPI/MediaStream.cc

```diff
@@ -381,18 +402,50 @@ void MediaStream::notifyStats(const std::string& message) {
   uv_async_send(async_stats_);
 }
 
+void MediaStream::notifyMediaStreamEvent(const std::string& type, const std::string& message) {
+  if (!this->hasEventCallback_) {
+    return;
+  }
+  if (!async_event_) {
+    return;
+  }
+  boost::mutex::scoped_lock lock(mutex);
+  this->eventMsgs.push(std::make_pair(type, message));
+  async_event_->data = this;
+  uv_async_send(async_event_);
+}
+
 NAUV_WORK_CB(MediaStream::statsCallback) {
   Nan::HandleScope scope;
   MediaStream* obj = reinterpret_cast<MediaStream*>(async->data);
   if (!obj || !obj->me) {
     return;
   }
   boost::mutex::scoped_lock lock(obj->mutex);
-  if (obj->hasCallback_) {
+  if (obj->hasStatsCallback_) {
```

Comment:
I now think we should block any access to hasStatsCallback_ and statsCallback_ with a mutex too, since they can be accessed from multiple threads (e.g. MediaStream::close, MediaStream::notifyMediaStreamEvent, and MediaStream::statsCallback). And the same happens to the rest of callbacks. We can do it in a separate PR though, wdyt?